### PR TITLE
[SYCL][Doc] Add FAQ section

### PR DESCRIPTION
**Pull request by alexbatashev**
This patch provides basic FAQ section with answers to the most
frequently asked questions. The list of questions is not complete and
will be extended by other patches.

Co-authored-by: Artem Gindinson <artem.gindinson@intel.com>
Signed-off-by: Alexander Batashev <alexander.batashev@intel.com>

### CODE REVIEW

File: sycl/doc/FAQ.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+exploit SYCL's host device feature. This way gives you ability to use any C++11
+compiler. You will need to link your application with SYCL Runtime library and
+provide path to SYCL headers directory. Please, refer to you compiler manual
+to learn about specific build options.
+
+### Q: How are SYCL compilation phases different from those of a usual C++ compiler? Can I customize this flow for my applications?
+**A:** Due to the fact that both host and device code need to be compiled and linked into the final binary, the compilation steps sequence
+is more complicated compared to the usual C++ flow.
+
+In general, we encourage our users to rely on SYCL Compiler for handling all of the compilation phases "under the hood". However, thorough
```
[the remark this answers is not in the document]
Comment (AGindinson) [reply sycl/doc/FAQ.md]: This one depends on the context. E.g. _"rely on IntellijIDEA"_ (no article, proper noun), but _"rely on an/the IDE"_ (a mere concrete noun). Here, we're kind of implying "our SYCL implementation", which is, technically, the first scenario.

File: sycl/doc/FAQ.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+1. [Common issues](#common-issues)
+1. [Device specific questions and issues](#device-specific-questions-and-issues)
+
+
+## Developing with SYCL
+
+### Q: What do I need to start developing with SYCL?
+**A:** To get full SYCL experience you need a SYCL-capable compiler. Intel SYCL
+compiler provides you with both host and device side compilation. Another requirement
+for device code offloading to accelerators is a compatible OpenCL runtime.
```
[the remark this answers is not in the document]
Comment (AGindinson) [reply sycl/doc/FAQ.md]: I've made an attempt to reword this.

File: sycl/doc/FAQ.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+Windows:
+
+![Error screen](images/missing_sycl_dll.png)
+
+*The code execution cannot proceed because sycl.dll was not found. Reinstalling the program may fix this problem.*
+
+**A:** The SYCL Runtime library is required to run SYCL-enabled applications. While
+compiler driver is able to find the library and link against it, your operating
+system may struggle. Make sure that the location of the SYCL runtime library is listed in
+`LD_LIBRARY_PATH` (for Linux) or `LIB` (for Windows) environment variables.
```
[the remark this answers is not in the document]
Comment (AGindinson) [reply sycl/doc/FAQ.md]: Reworded to avoid any ambiguities.

File: sycl/doc/FAQ.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+## Developing with SYCL
+
+### Q: What do I need to start developing with SYCL?
+**A:** To get full SYCL experience you need a SYCL-capable compiler. Intel SYCL
+compiler provides you with both host and device side compilation. Another requirement
+for device code offloading to accelerators is a compatible OpenCL runtime.
+Our [Get Started Guide](GetStartedWithSYCLCompiler.md) will help you set up proper
+environment. To learn more about SYCL compiler usage, please refer
+to [User Manual](SYCLCompilerUserManual.md). If using a special compiler is not
+an option for you and you don't need to offload code to accelerators, you can
```
[the remark this answers is not in the document]
Comment (AGindinson) [reply sycl/doc/FAQ.md]: Reworded so as to reflect this idea

**Comment by AGindinson** [comment]
Applied most of the suggestions, resolving the correspondent comments.

File: sycl/doc/FAQ.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (AGindinson) [reply sycl/doc/FAQ.md]: Reconsidered.

**Comment by AGindinson** [comment]
Hopefully, a SYCL LIT failure won't block _this_ PR...

**Comment by bader** [comment]
> Hopefully, a SYCL LIT failure won't block _this_ PR...

No, but I'd like to receive an approve from @pvchupin.

**Pull request merged**